### PR TITLE
[#172376369] Add test profile flag

### PR DESCRIPTION
**Pull request by AleDore**
Added isTestUser flag to Profile in order to enable login without SPID/CIE for testing purpose.

### CODE REVIEW

File: src/models/profile.ts [hunk 1]
```diff
@@ -68,7 +69,10 @@ export const Profile = t.intersection([
 
     // array of user's preferred languages in ISO-3166-1-2 format
     // https://it.wikipedia.org/wiki/ISO_3166-2
-    preferredLanguages: PreferredLanguages
+    preferredLanguages: PreferredLanguages,
+
+    // if true this profile is only for test purpose
+    isTestProfile: withDefault(IsTestProfile, false)
```
Comment (gunzip) [inline src/models/profile.ts:75]: can you check that the generated IsTestProfile has already a default value ?

**Comment by gunzip** [comment]
it looks like tests do not pass

**Comment by gunzip** [comment]
@Undermaken how does the app behave in case the profile has one more field ? does it still parses the json ?

**Comment by Undermaken** [comment]
> @Undermaken how does the app behave in case the profile has one more field ? does it still parses the json ?

@gunzip 
As long as the payload meets the specs in terms of required and optional fields, any other additional fields don't represent an issue in profile decoding.
Here a proof

```typescript
const strg = `{
  "accepted_tos_version": 1,
  "email": "mario.rossi@email.it",
  "family_name": "Rossi",
  "has_profile": true,
  "is_inbox_enabled": true,
  "is_email_enabled": true,
  "is_email_validated": true,
  "is_webhook_enabled": true,
  "another_field": "hello!",
  "name": "Mario",
  "spid_email": "mario.rossi@spid-email.it",
  "spid_mobile_phone": "555555555",
  "version": 2,
  "fiscal_code": "RSSMRA83A12H501D",
  "blocked_inbox_or_channels": {
    "dev-service_0": [
      "INBOX"
    ]
  }
}`;
console.log(InitializedProfile.decode(JSON.parse(strg)));
```
output
```typescript
Right {
  value: {
    accepted_tos_version: 1,
    email: 'mario.rossi@email.it',
    family_name: 'Rossi',
    has_profile: true,
    is_inbox_enabled: true,
    is_email_enabled: true,
    is_email_validated: true,
    is_webhook_enabled: true,
    another_field: 'hello!',
    name: 'Mario',
    spid_email: 'mario.rossi@spid-email.it',
    spid_mobile_phone: '555555555',
    version: 2,
    fiscal_code: 'RSSMRA83A12H501D',
    blocked_inbox_or_channels: { 'dev-service_0': [Array] }
  },
  _tag: 'Right'
} 
```

File: src/models/profile.ts (hunk 1 above)
Comment (AleDore) [reply src/models/profile.ts:75]: - check on tests in [aa27b03]()

- generated code : `
export type IsTestProfile = t.TypeOf<typeof IsTestProfileBase>;
const IsTestProfileBase = t.boolean;

export const IsTestProfile = withDefault(
  IsTestProfileBase,
  false as IsTestProfile
);
`

Comment (gunzip) [reply src/models/profile.ts:75]: as it already has a default value we must chagne this to

```suggestion
    isTestProfile: IsTestProfile
```

**Pull request merged**